src/coreclr/src/pal/src/map/virtual.cpp

```diff
@@ -2284,7 +2284,11 @@ int32_t ExecutableMemoryAllocator::GenerateRandomStartOffset()
     // This code is similar to what coreclr runtime does on Windows.
     // It generates a random number of pages to skip between 0...MaxStartPageOffset.
     srandom(time(NULL));
+#if HAVE_ARC4RANDOM_UNIFORM
+    pageCount = (int32_t)arc4random_uniform(MaxStartPageOffset);
```

Comment:
What's the point of `srandom(time(NULL))` here and what's wrong with `random()`?

Reply:
srandom is there to set the random generator seed based on the current time so that the random offset of executable memory allocations is not always the same (this code is invoked just once at process startup).
I think there is nothing wrong with random, the point was that we needed to differentiate between Illumos and other unixes we already support as its RAND_MAX is set to 0x7fff and it doesn't apply to the random() output (it does for rand() only). And since the arcrandom has a function that doesn't have to care about the RAND_MAX, it seemed like a clean replacement.

Reply:
You mean `srandom(time(NULL))` is not needed in `#if` branch and only applicable in case of `random()`? Right,  arc4random* does not require an initial seed (with srand or srandom). I will move it under else.

reason why we are using `arc4random_uniform` is because it is thread safe and we already use arc4random_* in src/libraries.

Reply:
OK, but arc4random_uniform does not need srandom and it looks like a leftover.

Reply:
This diff is not showing it for some reason, but if you look at `Files` view, it is moved under `#else`.

Reply:
We are basically replacing pseudo-random simple generator with a good random generator for other platforms. There will be mismatch with what we get on what OS. random() at least on NetBSD is thread safe... but it is not fork safe. If it is thread unsafe on other platforms, I recommend to use the same random generation for all platforms. If it is missing somewhere, add a fallback code that reimplements it (copy-pasting from BSD is enough)/.

I don't like mixing strong and weak random generators with different properties depending on type of OS.

Reply:
.

Reply:
Ok, I understand the concern, we can get back to using random everywhere and just use a different divisor (RAND_MAX or 0x7fffffff) depending on the platform.
The thread safety is not a concern here, this code is executed only once during PAL initialization when there is just one thread running.

Reply:
Thanks, I have changed it to use random() everywhere.